### PR TITLE
Support creating files with different extension than glsl, closes #72

### LINKED ISSUE #72: Can't create files with an extension other than .glsl
If I use 'Right click › New › GLSL Shader', and enter 'Shader.vert', it creates a new file called 'Shader.vert.glsl'. It still recognises it as a GLSL file, but I then have to manually rename it to 'Shader.vert'.

### PR DESCRIPTION
**Pull request by Darkyenus**
Also makes files with `frag` and `vert` extensions use different template. The template is however used only if the default `glsl` template has not been changed, then that changed one is used.

Problematic is, that these new templates cannot be changed. Ideally, there would be multiple templates and `GLSLCreateFromTemplateHandler` would choose the correct one dynamically, based on the extension. Right now it selects a hardcoded one.

This should be possible through internal templates, but I cannot figure out how to add them. Adding `<internalFileTemplate name="something"/>` to `plugin.xml` and moving the files to fileTemplates/internal directory does not seem to find them. If somebody figures that out, the rest will be easy. (It would also mean that the existing templates have a fallback and are editable normally)

This change also improves extension handling for glsl files.
If there is no extension, `glsl` is appended. Otherwise, existing extension is used and if it is not recognized, `.glsl` is additionally appended.

EDIT: Maybe I have found how to do it? I'll investigate more.

### CODE REVIEW

File: resources/templates/default.frag [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#version 120
+
+void main() {
+	glFragColor = vec4(1.0, 0.0, 0.0, 1.0);
+}
```
Comment (AbigailBuccaneer) [inline resources/templates/default.frag]: No newline at end of file - Git and GitHub complain when they're missing.

Comment (Darkyenus) [inline resources/templates/default.frag]: Will fix

**Comment by Darkyenus** [comment]
Somehow, I did exactly what I wanted. It is slightly more complicated, but here is how it works:
- Action was added to the new file menu. This action looks brings up familiar dialog for new file creation. This is needed, because all templates are internal and don't show up in the menu anymore.
- Dialog window delegates the file creation to `GLSLCreateFromTemplateHandler` which handles already mentioned shuffling with extension
- It also searches through defined templates, which are defined with extension. If there is a template defined specifically for that extension, it is used instead of the default one
- Then it just creates the file, done

That is it, ready to merge. Default templates can be edited and new ones can be added - they will be picked up by the system automagically. (Unless they are overriding the default templates, then they won't be used in automatic template system, but will still be available through new file menu as standard templates)

**Comment by AbigailBuccaneer** [comment]
Cool stuff! :+1:

**Pull request merged**